### ISSUE TITLE
Erroneous call to default ctor generated if the type declares a static constructor

### ISSUE THREAD

**Issue by justinth** [k94ll13nn3/AutoConstructor#71]
### Description

If a type decorated with the `[AutoConstructor]` attribute declares a static constructor, then the generated code contains a call to a default constructor that does not exist, causing the build to fail.

### Steps to reproduce
Add a static constructor as follows:
```
namespace ConsoleApp1;

[AutoConstructor]
internal partial class Foo
{
    static Foo()
    {
    }

    private readonly object _someDependency;
}
```

This will generate the following code which will not compile:
```
//------------------------------------------------------------------------------
// <auto-generated>
//     This code was generated by the AutoConstructor source generator.
//
//     Changes to this file may cause incorrect behavior and will be lost if
//     the code is regenerated.
// </auto-generated>
//------------------------------------------------------------------------------
namespace ConsoleApp1
{
    partial class Foo
    {
        public Foo(object someDependency) : this()
        {
            this._someDependency = someDependency;
        }
    }
}

```

### Details
- Version affected: 4.1.0 (regression)

**Comment by k94ll13nn3** [comment]
Thank you for the report, it seems that the tests around static classes and constructors are not good enough, I will try to have a fix this evening, in the meantime you can go back to 4.0.3, as the `this` call was introduced in `4.1.0` (and it is the only feature for this release).

**Comment by k94ll13nn3** [comment]
Fixed in `4.1.1`

**Comment by justinth** [comment]
Thank you very much for the fast turn around, I really like this project and use it extensively!
